File: src/org/stormgears/utils/dsio/DummyButton.java
package org.stormgears.utils.dsio;

import edu.wpi.first.wpilibj.command.Command;

public class DummyButton implements IButton {
	@Override
	public void whenPressed(Runnable callback) {

	}

	@Override
	public void whenPressed(Command command) {

	}

	@Override
	public void whileHeld(Command command) {

	}

	@Override
	public void whenReleased(Command command) {

	}

	@Override
	public void whenReleased(Runnable callback) {

	}

	@Override
	public void toggleWhenPressed(Command command) {

	}

	@Override
	public void cancelWhenPressed(Command command) {

	}

	@Override
	public void whenActive(Command command) {

	}

	@Override
	public void whileActive(Command command) {

	}

	@Override
	public void whenInactive(Command command) {

	}

	@Override
	public void toggleWhenActive(Command command) {

	}

	@Override
	public void cancelWhenActive(Command command) {

	}
}

File: src/main/java/org/stormgears/utils/dsio/DummyButton.kt
package org.stormgears.utils.dsio

import edu.wpi.first.wpilibj.command.Command

class DummyButton : IButton {

	override fun whenPressed(callback: Runnable) {

	}

	override fun whenPressed(command: Command) {

	}

	override fun whenPressed(callback: () -> Unit) {

	}

	override fun whileHeld(command: Command) {

	}

	override fun whenReleased(command: Command) {

	}

	override fun whenReleased(callback: Runnable) {

	}

	override fun whenReleased(callback: () -> Unit) {

	}

	override fun toggleWhenPressed(command: Command) {

	}

	override fun cancelWhenPressed(command: Command) {

	}

	override fun whenActive(command: Command) {

	}

	override fun whileActive(command: Command) {

	}

	override fun whenInactive(command: Command) {

	}

	override fun toggleWhenActive(command: Command) {

	}

	override fun cancelWhenActive(command: Command) {

	}

	override fun get(): Boolean {
		return false
	}
}
